detectron2/data/datasets/coco.py

```diff
@@ -387,36 +387,37 @@ def convert_to_coco_dict(dataset_name):
     return coco_dict
 
 
-def convert_to_coco_json(dataset_name, output_folder="", allow_cached=True):
+def convert_to_coco_json(dataset_name, output_file, allow_cached=True):
     """
     Converts dataset into COCO format and saves it to a json file.
-    dataset_name must be registered in DatastCatalog and in detectron2's standard format.
+    dataset_name must be registered in DatasetCatalog and in detectron2's standard format.
 
     Args:
         dataset_name:
             reference from the config file to the catalogs
-            must be registered in DatastCatalog and in detectron2's standard format
-        output_folder: where json file will be saved and loaded from
+            must be registered in DatasetCatalog and in detectron2's standard format
+        output_file: path of json file that will be saved to
         allow_cached: if json file is already present then skip conversion
     Returns:
         cache_path: path to the COCO-format json file
     """
 
     # TODO: The dataset or the conversion script *may* change,
     # a checksum would be useful for validating the cached data
-    cache_path = os.path.join(output_folder, f"{dataset_name}_coco_format.json")
-    PathManager.mkdirs(output_folder)
-    if os.path.exists(cache_path) and allow_cached:
-        logger.info(f"Reading cached annotations in COCO format from:{cache_path} ...")
+
+    PathManager.mkdirs(os.path.dirname(output_file))
+    if os.path.exists(output_file) and allow_cached:
+        logger.info(f"Cached annotations in COCO format already exist: {output_file}")
```

Comment:
At this point the file could be a partial file that's being written to. So we could add 
```
with file_lock(...):
    pass
```
to wait for the writing.

Assuming that the `output_file` is only going to be written by one process (with the comment below), this should be sufficient.